src/test/java/com/snowflake/kafka/connector/internal/streaming/schemaevolution/iceberg/ParseIcebergColumnTreeTest.java

```diff
@@ -0,0 +1,236 @@
+            "ARRAY(OBJECT(name VARCHAR, id LONG))"),
+        // array
+        arguments("{\"testColumnName\": [1,2,3] }", "ARRAY(LONG)"),
+        arguments("{ \"testColumnName\": [] }", "ARRAY(VARCHAR(16777216))"));
+  }
+
+  @ParameterizedTest
+  @MethodSource("mergeTestArguments")
+  void mergeTwoTreesTest(String plainIcebergSchema, String recordJson, String expectedResult) {
+    // given tree parsed from channel
+    Type type = IcebergDataTypeParser.deserializeIcebergType(plainIcebergSchema);
+    ApacheIcebergColumnSchema apacheSchema = new ApacheIcebergColumnSchema(type, "TESTSTRUCT");
+    IcebergColumnTree alreadyExistingTree = new IcebergColumnTree(apacheSchema);
+
+    // tree parsed from a record
+    SinkRecord record = createKafkaRecord(recordJson, false);
+    JsonNode recordNode = RecordService.convertToJson(record.valueSchema(), record.value(), true);
+    IcebergColumnJsonValuePair columnValuePair =
+        IcebergColumnJsonValuePair.from(recordNode.fields().next());
+
+    IcebergColumnTree modifiedTree = new IcebergColumnTree(columnValuePair);
+    // then
+    alreadyExistingTree.merge(modifiedTree);
+
+    String expected = expectedResult.replaceAll("/ +/g", " ");
+    Assertions.assertEquals(expected, alreadyExistingTree.buildType());
+    Assertions.assertEquals("TESTSTRUCT", alreadyExistingTree.getColumnName());
+  }
+
+  static Stream<Arguments> mergeTestArguments() {
+    return Stream.of(
+        arguments(
+            "{\"type\":\"struct\",\"fields\":[{\"id\":23,\"name\":\"k1\",\"required\":false,\"type\":\"int\"},{\"id\":24,\"name\":\"k2\",\"required\":false,\"type\":\"int\"}]}",
+            "{ \"testStruct\": { \"k1\" : 1, \"k2\" : 2, \"k3\" : 3 } }",
+            "OBJECT(k1 INT, k2 INT, k3 LONG)"),
+        arguments(
+            "{\"type\":\"struct\",\"fields\":["
+                + "{\"id\":23,\"name\":\"k1\",\"required\":false,\"type\":\"int\"},{\"id\":24,\"name\":\"k2\",\"required\":false,\"type\":\"int\"},"
+                + "  {\"id\":25,\"name\":\"nested_object\",\"required\":false,\"type\":{\"type\":\"struct\",\"fields\":["
+                + "    {\"id\":26,\"name\":\"nested_key1\",\"required\":false,\"type\":\"string\"},"
+                + "    {\"id\":27,\"name\":\"nested_key2\",\"required\":false,\"type\":\"string\"}"
+                + "]}}]}",
+            "{\"testStruct\" : {"
+                + "  \"k1\" : 1, "
+                + "  \"k2\" : 2, "
+                + "    \"nested_object\": { "
+                + "      \"nested_key1\" : \"string\", "
+                + "      \"nested_key2\" : \"blah\", "
+                + "      \"nested_object2\" : { "
+                + "        \"nested_key2\" : 23.5 "
+                + "    }}"
+                + "}}",
+            "OBJECT(k1 INT, k2 INT, nested_object OBJECT(nested_key1"
+                + " VARCHAR(16777216), nested_key2 VARCHAR(16777216), nested_object2"
+                + " OBJECT(nested_key2 DOUBLE)))"),
```

Comment:
we should consider keeping the schema in separate txt files and the messages in json files as this is becoming really difficult to read